### PR TITLE
Auto build client appimage

### PR DESCRIPTION
**Pull request by Lejo1**
Appimages are easy to use cross compatible files including everything to run the application.
This PR adds an automated build to create a client appimage using appimagebuilder to the gitlab ci

Such appimages are portable and don't require any additional installations.

Thanks to @An0n3m0us and @Calinou for the helping me out!

PR also removes the old ubuntu:14-04 build with the new ubuntu:18-04 build

## To do

This PR is ready to review. Might need some library changes for OS support

- [ ] Test on different linux distributions (Done: Ubuntu, Debian)
- [x] Resolve the Problem in the Desktop file (The builder complains about PrefersNonDefaultGPU=true)
- [x] Clean library list

## How to test

I created a [gitlab Ci](https://gitlab.com/lejo1/minetest/-/pipelines) for my minetest fork.
[Download Appimage](https://gitlab.com/lejo1/minetest/-/jobs/823672224/artifacts/download)

### CODE REVIEW

**Comment by Calinou** [comment]
>  Resolve the Problem in the Desktop file (The builder complains about PrefersNonDefaultGPU=true)

You can remove `PrefersNonDefaultGPU=true` from the desktop file using `sed` when building.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -308,3 +308,28 @@ pages:
   only:
     - master
 
+package:appimage-client:
+  stage: package
+  image: appimagecrafters/appimage-builder
+  dependencies:
+    - build:ubuntu-16.04
```
Comment (sfan5) [inline .gitlab-ci.yml:315]: why so outdated?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -308,3 +308,28 @@ pages:
...
+  before_script:
+    - apt-get update -y
+    - apt-get install -y git wget rsync
+    # Collect files
+    - mkdir AppDir
+    - rsync -a artifact/minetest/usr/ AppDir/usr/
```
Comment (sfan5) [inline .gitlab-ci.yml:321]: you could use `cp -a` here

File: AppImageBuilder.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - sourceline: deb http://archive.ubuntu.com/ubuntu/ bionic-updates main universe
+      - sourceline: deb http://archive.ubuntu.com/ubuntu/ bionic-backports main universe
+      - sourceline: deb http://archive.ubuntu.com/ubuntu/ bionic-security main universe
+
+# Included stuff: libirrlicht-dev libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
+    include:
+      - libirrlicht1.8
+      - libbz2-1.0
+      - libpng16-16
+      - libjpeg8
```
Comment (sfan5) [inline AppImageBuilder.yml:31]: transitive irricht deps, not needed here

File: AppImageBuilder.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      - libsqlite3-0
+      - libogg0
+      - libvorbis0a
+      - libopenal1
+      - libcurl3-gnutls
+      - libfreetype6
+      - zlib1g
+      - libgmp10
+      - libjsoncpp1
+      - libfuse2
```
Comment (sfan5) [inline AppImageBuilder.yml:43]: FUSE is not needed

**Comment by Lejo1** [comment]
> You can remove PrefersNonDefaultGPU=true from the desktop file using sed when building.

Thanks, Good Idea! Did that.

File: .gitlab-ci.yml (hunk 1 above)
Comment (Lejo1) [reply .gitlab-ci.yml:315]: I didn't wanted to add another build so I recycled one. It's the newest ubuntu build in the gitlab ci.
I agree it should be updated but that's just not the goal of this PR

File: .gitlab-ci.yml (hunk 2 above)
Comment (Lejo1) [reply .gitlab-ci.yml:321]: Right, didn't knew how to do that correctly. Google said rsync :D

File: AppImageBuilder.yml (hunk 3 above)
Comment (Lejo1) [reply AppImageBuilder.yml]: Changed, thanks!

File: AppImageBuilder.yml (hunk 4 above)
Comment (Lejo1) [reply AppImageBuilder.yml]: Changed, thanks!

Comment (Lejo1) [reply AppImageBuilder.yml]: Changed, thanks!

File: .gitlab-ci.yml (hunk 1 above)
Comment (sfan5) [reply .gitlab-ci.yml:315]: 14.04 should be removed in favor of 18.04 then.
Please do that in a separate commit so it can be kept apart on merge.

Comment (Lejo1) [reply .gitlab-ci.yml]: Did that.
Build is just like in ubuntu:16-04

**Comment by An0n3m0us** [comment]
`Ubuntu 16.04: libunistring.so.2: cannot open shared object file: No such file or directory`
After adding that library, the game works as expected.

I would fix it myself if I knew how these automated builds work xD

**Comment by Lejo1** [comment]
Thanks for the feedback!
I excluded some random libraries cause I thought they weren't needed (also libunistring).
Should work now (When build is done and I changed the link)

File: AppImageBuilder.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  apt:
+    arch: amd64
+    sources:
+      - sourceline: deb http://archive.ubuntu.com/ubuntu/ bionic main universe
+        key_url: 'http://keyserver.ubuntu.com/pks/lookup?op=get&search=0x3b4fe6acc0b21f32'
+      - sourceline: deb http://archive.ubuntu.com/ubuntu/ bionic-updates main universe
+      - sourceline: deb http://archive.ubuntu.com/ubuntu/ bionic-backports main universe
+      - sourceline: deb http://archive.ubuntu.com/ubuntu/ bionic-security main universe
+
+# Included stuff: libirrlicht-dev libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
```
Comment (sfan5) [inline AppImageBuilder.yml:26]:
```suggestion
# Included stuff: libirrlicht-dev libbz2-dev libpng-dev libjpeg-dev libxxf86vm-dev libgl1-mesa-dev libsqlite3-dev libogg-dev libvorbis-dev libopenal-dev libcurl4-gnutls-dev libfreetype6-dev zlib1g-dev libgmp-dev libjsoncpp-dev
```

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -308,3 +303,28 @@ pages:
...
+    - mkdir AppDir
+    - cp -a artifact/minetest/usr/ AppDir/usr/
+    - rm AppDir/usr/bin/minetestserver
+    - cp -R clientmods AppDir/usr/share/minetest
+  script:
+    - git clone $MINETEST_GAME_REPO AppDir/usr/share/minetest/games/minetest_game
+    - rm -Rf AppDir/usr/share/minetest/games/minetest/.git
+    - export VERSION=$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA
+    # Remove PrefersNonDefaultGPU=true
+    - sed -i '/PrefersNonDefaultGPU=true/d' AppDir/usr/share/applications/net.minetest.minetest.desktop
```
Comment (sfan5) [inline .gitlab-ci.yml:324]:
```suggestion
    # Remove PrefersNonDefaultGPU property due to validation errors
    - sed -i '/PrefersNonDefaultGPU/d' AppDir/usr/share/applications/net.minetest.minetest.desktop
```

**Review by sfan5 — APPROVED** [review]
Looks good.

**Review by SmallJoker — APPROVED** [review]
Yes sure, why not. LGTM.

**Pull request merged**